pkg/networking/cilium/templater.go

```diff
@@ -61,8 +60,16 @@ func (t *Templater) GenerateUpgradePreflightManifest(ctx context.Context, spec *
 	if err != nil {
 		return nil, err
 	}
+	helm, err := t.helmFactory.GetClientForCluster(ctx, spec.Cluster)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get helm client for cluster %s: %v", spec.Cluster.ManagedBy(), err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("getting helm client for cluster %s: %v", spec.Cluster.Name, err)
```

the abstraction should hide the fact that the client is for the management cluster

Reply:
Okay, I'll adjust these